crates/lib3h/src/transport/protocol.rs

```diff
@@ -1,66 +1,44 @@
-use crate::{
-    lib3h_protocol::data_types::Opaque,
-    transport::{error::TransportError, ConnectionId},
-};
+use crate::transport::error::TransportError;
 use lib3h_ghost_actor::prelude::*;
+use lib3h_protocol::data_types::Opaque;
 use url::Url;
-/// Commands that can be sent to an implementor of the Transport trait and handled during `process()`
-#[derive(Debug, PartialEq, Clone)]
-pub enum TransportCommand {
-    Connect(Url, /*request_id*/ String),
-    Send(Vec<ConnectionId>, Opaque),
-    SendAll(Opaque),
-    Close(ConnectionId),
-    CloseAll,
-    Bind(Url),
-}
 
-/// Events that can be generated during a `process()`
-#[derive(Debug, PartialEq, Clone)]
-pub enum TransportEvent {
-    /// Notify that some TransportError occured
-    ErrorOccured(ConnectionId, TransportError),
-    /// an outgoing connection has been established
-    ConnectResult(ConnectionId, /*request_id*/ String),
-    /// we have received an incoming connection
-    IncomingConnectionEstablished(ConnectionId),
-    /// We have received data from a connection
-    ReceivedData(ConnectionId, Opaque),
-    /// A connection closed for whatever reason
-    ConnectionClosed(ConnectionId),
+#[derive(Debug, Clone)]
+pub struct BindResultData {
+    pub bound_url: Url,
 }
 
 /// Transport protocol enums for use with GhostActor implementation
 #[derive(Debug, Clone)]
 pub enum RequestToChild {
     Bind { spec: Url }, // wss://0.0.0.0:0 -> all network interfaces first available port
-    SendMessage { address: Url, payload: Opaque },
-}
-
-#[derive(Debug, Clone)]
-pub struct BindResultData {
-    pub bound_url: Url,
+    SendMessage { uri: Url, payload: Opaque },
 }
 
 #[derive(Debug, Clone)]
 pub enum RequestToChildResponse {
     Bind(BindResultData),
-    SendMessage,
+    SendMessage { payload: Opaque },
 }
 
 #[derive(Debug, Clone)]
 pub enum RequestToParent {
-    IncomingConnection { address: Url },
-    ReceivedData { address: Url, payload: Opaque },
-    TransportError { error: TransportError },
+    ErrorOccured { uri: Url, error: TransportError },
```

Comment:
We shouldn't need the uri here. If you get an error during a `SendMessage` for example, you will send an `Err` result back with the given request_id. If a connection is dropped for any reason between calls, I think it can be ignored. When you again try to send data on that connection, it should automatically re-connect or send back an error, again on the `SendMessage` response.

This error without the uri was intended for things like the bound network interface gets disconnected. (Hot swap-able NIC?)